code/sys/sys_unix.c

```diff
@@ -552,7 +553,10 @@ void Sys_Sleep( int msec )
 		if( msec < 0 )
 			msec = 10;
 
-		usleep( msec * 1000 );
+		struct timespec req;
+		req.tv_sec = msec/1000;
+		req.tv_nsec = (msec%1000)*1000000;
+		nanosleep(&req, NULL);
```

Comment:
Note: `nanosleep` can be interrupted by a signal handler (if other than 0 value is returned), so if this function needs always to wait for exact time period, should be called again. In this case you could use its second argument.

https://www.man7.org/linux/man-pages/man2/nanosleep.2.html

Reply:
I think we can ignore the return value since the previous code ignored the value of the call to `usleep(3)`, so it looks like we don't care whether or not we waited for the exact time period.